larray_editor/editor.py

```diff
@@ -101,16 +106,30 @@ def setup_and_check(self, data, title='', readonly=False, **kwargs):
     def setup_menu_bar(self):
         """Setup menu bar"""
         menu_bar = self.menuBar()
-        self._setup_file_menu(menu_bar)
-        self._setup_edit_menu(menu_bar)
-        self._setup_help_menu(menu_bar)
+        if self._file_menu:
+            self._setup_file_menu(menu_bar)
+        if self._edit_menu:
+            self._setup_edit_menu(menu_bar)
+        if self._help_menu:
+            self._setup_help_menu(menu_bar)
 
     def _setup_file_menu(self, menu_bar):
         file_menu = menu_bar.addMenu('&File')
         file_menu.addAction(create_action(self, _('&Quit'), shortcut="Ctrl+Q", triggered=self.close))
 
     def _setup_edit_menu(self, menu_bar):
-        pass
+        if qtconsole_available:
```

Comment:
why do you need this test? the undo/redo functionality requires ipython? I don't see any fundamental reason why it should be so, but I might be missing something.

Reply:
If you go to module `commands.py`, you will see that what `Undo` and `Redo` does is to generate the right command and then execute `self.editor.kernel.shell.run_cell(command)`